### ISSUE TITLE
Support for right to left languages

### ISSUE THREAD

**Issue by MortenHofft** [reakit/reakit#546]
### Motivation

I've just started experimenting with this library a few days and I love the way it is implemented, the focus on a11y and the unstyled approach 👏 .  But unfortunately support for rtl languages (say arabic) is missing.

### Example
When setting `dir="rtl"` on an element, then actions on children elements should invert the concept of left right (NB: I'm not a rtl user).

```
// initial focus on Button 1. pressing left arrow will now shift focus to button 2.
<div dir="rtl">
  <Group>
      <Rover {...rover} as={Button}>
        Button 1
      </Rover>
      <Rover {...rover} as={Button}>
        Button 2
      </Rover>
      <Rover {...rover} as={Button}>
        Button 3
      </Rover>
    </Group>
</div>
```

### Possible implementations
I guess the [direct association](https://github.com/reakit/reakit/blob/master/packages/reakit/src/Rover/Rover.ts#L113) between right and next needs to be dependent on nearest element with an explicit direction stated (defaulting to ltr).

I noticed a similar issue at https://github.com/reach/reach-ui/issues/235

Thanks

**Comment by diegohaz** [comment]
Hi @MortenHofft! Thanks for raising this issue.

I have zero experience with RTL. There are plans to replace `rover.next()` and `rover.previous()` by `rover.right()` and `rover.left()` though. Would it solve this problem? What about `rover.first()` and `rover.last()`?

That said, you can also set the initial focus to the last element (although it will remain the last element):

```jsx
  const rover = useRoverState({ currentId: "last" });
  <Rover {...rover} as={Button}>
    Button 1
  </Rover>
  <Rover {...rover} as={Button} disabled>
    Button 2
  </Rover>
  <Rover {...rover} as={Button} disabled focusable>
    Button 3
  </Rover>
  <Rover {...rover} as={Button}>
    Button 4
  </Rover>
  <Rover {...rover} as={Button} stopId="last">
    Button 5
  </Rover>
```

**Comment by MortenHofft** [comment]
I would think that first and last would work just the same (pageUp and pageDown). It is simply that moving focus right when pressing left is odd.

instead of `ArrowRight: options.orientation !== "vertical" && options.next` in https://github.com/reakit/reakit/blob/master/packages/reakit/src/Rover/Rover.ts#L113 I would think it needs to evaluate the (current) direction of the target before deciding wether right means next or previous.
```
ArrowRight: options.orientation !== "vertical" && (e => {
  getComputedStyle(e.target).direction === 'rtl' ? options.previous() : options.next()
})
```

**Comment by l3tang** [comment]
Another way to do it would be to do the check for RTL inside of previous and next so that it doesn't need to be calculated everywhere.

**Comment by diegohaz** [comment]
As of `v1.0.0-rc.0`, `useTabState()`, `useMenuState()`, `useToolbarState()` and `useRadioState()` support a boolean `rtl` state.

We can discuss how to detect it automatically in a performant way in the future if needed, but for now I think we can close this.
